DuckDuckGo/TipKit/TipGroup.swift

```diff
@@ -0,0 +1,144 @@
+//
+//  TipGroup.swift
+//  DuckDuckGo
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import TipKit
+
+protocol TipGrouping {
+    @available(iOS 17.0, *)
+    @MainActor
+    var currentTip: (any Tip)? { get }
+}
+
+@available(iOS 18.0, *)
+extension TipGroup: TipGrouping {}
+
+/// A glorified no-op to be able to compile TipGrouping in iOS versions below 17.
+///
+struct EmptyTipGroup: TipGrouping {
+    @available(iOS 17.0, *)
+    var currentTip: (any Tip)? {
+        return nil
+    }
+}
```

Comment:
This is the struct to instantiate when the OS is lower than iOS 17.  It's a glorified no-op.

`currentTip` is guarded to iOS 17 because we really don't want it used at all, but is necessary in the protocol.